### PR TITLE
Add attach feature to Disk Model

### PR DESCRIPTION
**Pull request by noralife**
This patch enables disk model to attach to Server.
Please kindly review this.

### CODE REVIEW

File: lib/fog/sakuracloud/models/volume/disk.rb [hunk 1]
```diff
@@ -38,6 +38,11 @@ def carve_hostname_on_disk(hostname)
           true
         end
 
+        def attach(server_id)
+          service.attach_disk(@attributes[:id], server_id )
```
Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: You could just use `service.attach_disk(id, server_id)` here.

Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: No need to use the instance variable `@attributes` since each attribute has it own getter.

Comment (noralife) [inline lib/fog/sakuracloud/models/volume/disk.rb]: Thank you for reviewing.

I followed the same style above.
It uses @attributes a lot.
Do I use `id` just for this method?

```
        def save
          requires :name, :plan, :source_archive
          data = service.create_disk(@attributes[:name], @attributes[:plan], @attributes[:source_archive]).body["Disk"]
          merge_attributes(data)
          true
        end

        def configure(sshkey_id)
          requires :id
          service.configure_disk(@attributes[:id], sshkey_id )
          true
        end

        def carve_hostname_on_disk(hostname)
          requires :id
          service.carve_hostname_on_disk(@attributes[:id], hostname )
          true
        end

        def attach(server_id)
          service.attach_disk(@attributes[:id], server_id )
          true
        end
```

Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: Yeah, do it for only this method. We can refactor the rest in another PR.

File: lib/fog/sakuracloud/models/volume/disk.rb [hunk 2]
```diff
@@ -38,6 +38,11 @@ def carve_hostname_on_disk(hostname)
           true
         end
 
+        def attach(server_id)
+          service.attach_disk(id, server_id)
+          true
```
Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: Shouldn't you wait for the response of the request and then return instead of always return true?

Comment (noralife) [inline lib/fog/sakuracloud/models/volume/disk.rb]: I agree with you.
But please see other methods I pasted below. All methods return TRUE.
Just I follow the same style...

I think we need another refactoring for this model.

```
  class SakuraCloud
      class Disk < Fog::Model
...
        def save
          requires :name, :plan, :source_archive
          data = service.create_disk(@attributes[:name], @attributes[:plan], @attributes[:source_archive]).body["Disk"]
          merge_attributes(data)
          true
        end

        def configure(sshkey_id)
          requires :id
          service.configure_disk(@attributes[:id], sshkey_id )
          true
        end

        def carve_hostname_on_disk(hostname)
          requires :id
          service.carve_hostname_on_disk(@attributes[:id], hostname )
          true
        end

        def associate_ip(ipaddress, networkmasklen, defaultroute)
          subnet ={
            :ipaddress => ipaddress,
            :networkmasklen => networkmasklen,
            :defaultroute => defaultroute
          }
          requires :id
          service.associate_ip_to_disk(@attributes[:id], subnet )
          true
        end
```

Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: Yeah. Makes sense. It would be annoying if the request failed and it returned true. There isn't an easy way to check the result of the request so this is bad.

And i agree we should do this in another PR.

Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: Yeah. Makes sense. It would be annoying if the request failed and it returned true. There isn't an easy way to check the result of the request so this is bad.

And i agree we should do this in another PR.

EDIT: Maybe you could add a test around it?

@sawanoboly Please merge this as soon as you think it is good to go.

Thanks!

Comment (noralife) [inline lib/fog/sakuracloud/models/volume/disk.rb]: Currently fog-sakuracloud focuses on **requests** part as a test target.
So I think we should create another PR to add testcases for Model also.

Here is current contents inside tests dir.
```
tests/
├── helper.rb
└── sakuracloud
    └── requests
        ├── compute
        │   ├── plans_tests.rb
        │   ├── servers_tests.rb
        │   ├── ssh_keys_tests.rb
        │   └── zones_tests.rb
        ├── network
        │   ├── routers_tests.rb
        │   └── switches_tests.rb
        └── volume
            ├── archives_tests.rb
            ├── disks_tests.rb
            └── plans_tests.rb
```

Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: Yeah. Thats a good point. I like to have too business logic in models so i can test it without reaching the provider. This makes it easier to maintain.
Usually the request is just a thin layer on top of the request, doing encoding and parsing also. =)

**Comment by sawanoboly** [comment]
Thanks! I'll add usage to wiki.

**Comment by sawanoboly** [comment]
Released as v1.4.0.
